doc/MiniLanguageTutorial.md

```diff
@@ -112,31 +139,56 @@ Mini has the following types:
 
 > an untagged union type that can hold values from any of *type1*, *type2*, etc. There must be explicit casts to convert to and from its component types. This is done via `newunion` and `unioncast`.
 
-`struct` { *name1: type1 , name2 : type2 , ...* }
+`struct` `{` *name1* `:` *type1* `,` *name2* `:` *type2* `,` ... `}`
 
 > a struct with one or more named, typed fields (a compound type)
 
 `option`< *type* >
 
-> either the contained type or "None<*type*>", must be unwrapped to compare with inner type
+> Either `Some(`*value*`)` or `None<`*type*`>`, where *value* represents an arbitrary value of type *type*. 
+> The specific variant being used can be determined by `if let` and `?`.
+> The shortcut `None` can be used to create a value of `None<every>`, which is assignable to all `None<`T`>` for any type T.
+> The term "inner type" refers to the type specified by *type*.
+
+[ `view` | `write` | `public` ]* `func` ( *type1, type2, ...*) (-> *returntype*)?
+
+[ `view` | `write` ] `closure` `(` *type*, *type*, ... `)` (`->` *returntype*`)?
 
-[`impure`] `func` ( *type1, type2, ...*) [-> *returntype*]
+> A reference to a function, while `public` can be included a type declaration, it has no effect on the internal type.
 
-> a reference to a function
+*ident*
+
+> A nominal type or a type variable. It is only treated as a type variable in the context of a generic type or function,
+> and only if that identifier is listed as one of the type variables of that generic type or function. 
+> Otherwise, it is treated as a nominal type, if that nominal type is not defined in the file or is not imported with a use statement, 
+> it is not a valid type and will cause an error.
+> The representation of a nominal type is the type on the left side of the type declaration with the same name as *ident*.
+
+*ident* `<` *type1* `,` *type2* `,` ... `>`
+
+> A specialization of a generic type. *ident* must be the name of some generic type, and *type1*, *type2*, ...
+> and *type1*, *type2*, etc, replace each type variable of *ident* in order of their definition.
+> The representation of a specialized generic type is the type on the left side of the generic type declaration with the same name as *ident*.
 
 `any`
 
-> a value of unknown type
+> Can hold values of arbitrary type.
 
 `every`
 
-> a type that has no values, so that no value of this type can ever exist
+> A type that has no values. No value of this type can ever exist.
+
+### Generic types
+
+> Generic types that have not been specialized have no rule by which they can be parsed as types. However, internally they can be represented as types.
+> Therefore it is possible that one might appear if a bug in the compiler is present. In this case they will be be displayed as `generic` *integer*.
```

Comment:
I added this because of a comment in the `default_value` function, apparently this can appear in builtins, although I haven't seen any specific ways to trigger it: [ast.rs](https://github.com/OffchainLabs/arb-os/blob/documentation/src/compile/ast.rs#L860)

I'll check with Rachel later about a specific trigger.